nvm.sh

```diff
@@ -3227,11 +3229,13 @@ nvm_auto() {
       nvm install >/dev/null
     fi
   elif [ "_$NVM_MODE" = '_use' ]; then
-   VERSION="$(nvm_resolve_local_alias default 2>/dev/null || nvm_echo)"
-    if [ -n "$VERSION" ]; then
-      nvm use --silent "$VERSION" >/dev/null
-    elif nvm_rc_version >/dev/null 2>&1; then
-      nvm use --silent >/dev/null
+    if [ "_$NVM_CURRENT" = '_system' ]; then
```

Comment:
Indeed, that's the idea. The fact that NVM assumes, whenever it's sourced, that the environment is uninitialized is a hindrance to me.

(I've pushed a fix to handle `none`.)